docs/CONTRIBUTING.md

```diff
@@ -357,14 +380,51 @@ You can run individual unit tests by using the following syntax:
 Or by adding `.only` after a `describe` or `it` block in the test file:
 
 ```typescript
-describe.only('Extension manifest tests', () => {
-  it('should register all commands', () => {
+describe.only("Extension manifest tests", () => {
+  it("should register all commands", () => {
     // test code
   });
 });
 ```
 
-Remember to remove `.only` after running the test and before making a PR! 
+Remember to remove `.only` after running the test and before making a PR!
+
+#### Clicktesting the VS Code extension against the sidecar
+
+##### Method 1:
+
+Manually trigger the "Multi Arch Builds and Upload to GitHub Releases" promotion in the sidecar
+repo. This can be found in the Semaphore UI under the "Promotions" tab. This won't actually upload
+to a release, but will make the native executables available at the workflow's artifacts/ part of
+the Semaphore UI. Then you can continue on from step 3 of Method 2.
+
+##### Method 2:
+
+1. Checkout branch you want to test against in the sidecar.
+
+2. Run `make clean mvn-package-native-no-tests` in the sidecar to get the native executable packaged
+   up in the target directory.
+
+3. Copy the native executable from the sidecar's `target` directory to the `vscode/bin/` directory
+   (you can use click & drag).
+
+4. In vscode's .versions/ide-sidecar.txt make sure it matches that specified version in the native
+   executable file name.
+
+5. If there's a change to the REST or graphql model, grab the openapi generated file from sidecar
+   (underlying logic change doesn't require this step)
+
+6. Copy/paste that file to vscode's openapi file in
+   [`src/clients/sidecar-openapi-specs/sidecar.openapi.yaml`](../src/clients/sidecar-openapi-specs/sidecar.openapi.yaml)
+   or the schema in [`src/graphql/sidecar.graphql`](../src/graphql/sidecar.graphql).
+
+7. Run `gulp build`.
+
+8. Confirm client code changes are expected.
+
+9. Check that ts is still linting properly: `gulp lint`.
+
+10. Start the extension dev host.
```

Comment:
Thanks a lot for adding this documentation. It's extremely helpful.

In addition it, WDYT of adding a script to this repo that accepts the path to a local copy of the ide-sidecar repo as input parameter and automates these steps? It could build the sidecar, copy the native executable and the files from the `src/generated/resources` folder to the vs code extension project, re-generate the sidecar client, and generate the vsix file?

Reply:
@Cerchie it's probably best kept in here or `ide-sidecar` (or both) for easy access to internal and external contributors since `dtx-tools` is private.